src/index.ts

```diff
@@ -0,0 +1,48 @@
+// @ts-ignore
+import { isVue3, inject } from 'vue-demi';
+import {
+  createClient,
+  SupabaseClient,
+  SupabaseClientOptions,
+  SupabaseRealtimePayload,
+  AuthUser,
+  AuthSession,
+} from '@supabase/supabase-js';
+
+const supabaseSymbol = Symbol('supabase');
+
+/**
+ * Used to get the injected instance of SupabaseClient.
+ * If using Vuejs 2.x, this function will be undefined, please use
+ * `this.$supabase` instead.
+ * @returns SupabaseClient
+ */
+export function useSupabase(): SupabaseClient { 
+  return inject(supabaseSymbol);
+}
+
+export function install(app: any, options: any) {
```

Comment:
Anything we can do for the types here?

Reply:
There probably is, I'll check with the vue 2 and 3 codebase for what they pass into this method. For the options one, I think the SupabaseClientOptions should be the type